src/UICrafter.Core/Components/Render/ComponentListRender.razor

```diff
@@ -0,0 +1,51 @@
+
+@if (componentList != null)
+{
+	@foreach (var component in componentList.UIComponents)
+	{
+		@switch (component.ComponentCase)
+		{
+			case UIComponent.ComponentOneofCase.Button:
+				<RenderAPIButton Button="component" />
+				break;
+			case UIComponent.ComponentOneofCase.InputField:
+				<RenderInputField InputField="component" />
+				break;
+			case UIComponent.ComponentOneofCase.Textbox:
+				<RenderTextBox Textbox="component" />
+				break;
+			default:
+				break;
+		}
+	}
+} else if (UIComponents != null)
+{
+	@foreach (var component in UIComponents)
+	{
+		@switch (component.ComponentCase)
+		{
+			case UIComponent.ComponentOneofCase.Button:
+				<RenderAPIButton Button="component" />
+				break;
+			case UIComponent.ComponentOneofCase.InputField:
+				<RenderInputField InputField="component" />
+				break;
+			case UIComponent.ComponentOneofCase.Textbox:
+				<RenderTextBox Textbox="component" />
+				break;
+			default:
+				break;
+		}
+	}
+}
+
+
+
+
+@code{
+    [Parameter]
+    public UIComponentList? componentList { get; set; }
+
+    [Parameter]
+    public RepeatedField<UIComponent>? UIComponents { get; set; }
+}
```

Comment:
Just keep public RepeatedField<UIComponent>? UIComponents { get; set; } and then pass componentList.UIComponents instead of having 2 different params